### ISSUE TITLE
A new way to contribute rules (not the same way as Rule Recipes)

### ISSUE THREAD

**Issue by ReporterX** [uBlockOrigin/uMatrix-issues#131]
(Note: I didn't explain clearly for the first time. I rewrote it)

My suggestion is to offer a **new extra** way to contribute and share rules. They can complement each other.

There will be two ways: 
1. Ruleset recipes
2. Normal rules (New way)

To be clear, what "normal rules" mean here is [something like this](https://github.com/uMatrix-Rules/uMatrix-Rules-Domain/blob/master/rules.txt). In other words, it is the same as what you see in "my rules". You can simply contribute by uploading your list of rules.

Both mode have different purpose and can fulfill different needs.

![Two mode compare](https://user-images.githubusercontent.com/8235338/56077937-ac06e280-5e14-11e9-8012-12d86d4010a4.png)

### "Rule recipes" need constant user interaction

The original idea of rule recipes is great. Users have choices. It is good for power users who want to take things into control and fine-tune the rules.

However it requires constant user interaction. The rules do not apply automatically. Users have to review each time and apply manually.

For the masses, they want a quicker and more seamless way of browsing the Internet, enjoying the security and privacy benefits **with minimal user interaction**. Recipe could not fulfill such needs.

### Rules have to be rewritten before contribution

Existing rules have to be rewritten as "Rule recipe" formats before one can contribute their rules. This is a barrier and will hinder the growth of "Rule recipe".

It appears the development of [the official recipe](https://github.com/uBlockOrigin/uAssets/tree/master/recipes) is stagnant. The list is still very short (331 lines only as of 13 Apr 2019).

### New way can cut down user interaction and boost user base 

Let's take a look at Privacy Badge. This addon sets up a long list on behalf of the users. The user interaction is minimal. A lot more people installed and used it.

As of today (13 Apr 2019),  
Privacy Badge: 570,137 Users
uMatrix: 96,687 Users

To improve and boost the usage of uMatrix, I think we really need to cut down the level of user interaction. Boost more installation and we can protect more people.

### My suggestion: Normal rules (New way of contribution)

We need a new section to contribute "normal rules". There will be several sections: 
Hosts files  
Normal rules  
Ruleset recipes  

There are several benefits:  
1. No rewrite is needed before contributing your rules
2. "Set and forget", just like Hosts files

It is just one-click subscription, instead of asking users to review to apply (in "Ruleset recipes").

Encourage people to submit their own rules, and we have a central place to maintain a list of them. Users are free to subscribe any list of rules they want.

Set and forget. The masses like a lot.

We could have something like **Top 500 sites**, to help users to unbreak sites and offer security/privacy automatically. No more user interaction.

If we significantly reduce the barrier and the level of user interaction, the user base is going to skyrocket.

My two cents. Thank you for reading.

**Comment by uBlock-user** [comment]
Closing this because this is neither an issue/nor a suggestion. Discussion can continue though.

**Comment by gorhill** [comment]
My stance is that whoever is free to create such list of recipes, the same way Fanboy's lists were developed outside of ABP. If a list become prominent enough and there is a consensus regarding it's quality, I am opened to add it as a stock list in uMatrix.

**Comment by ReporterX** [comment]
> 
> 
> Closing this because this is neither an issue/nor a suggestion. Discussion can continue though.

Maybe I did not express clearly. The suggestion is actually having a new section called "preset rules". "Preset rules" is one-click subscription of all rules.

It is not the same as ruleset recipe. Ruleset recipe requires user interaction. You have to add it one by one when you encounter it.

**Comment by ReporterX** [comment]
> 
> 
> My stance is that whoever is free to create such list of recipes, the same way Fanboy's lists were developed outside of ABP. If a list become prominent enough and there is a consensus regarding its quality, I am opened to add it as a stock list in uMatrix.

I have several thousand lines of rules now. It is hard to convert all of them to recipes. It needs a complete re-write. Even then it does not help to reduce the user interaction.

That's why I suggest having a new section called "preset rules".  Preset rules are set and forget. The masses like a lot.

**Comment by uBlock-user** [comment]
Preset rules are ruleset recipes made by gorhill on purpose -- see - https://github.com/gorhill/uMatrix/issues/30

**Comment by ReporterX** [comment]
> 
> 
> Preset rules are ruleset recipes made by gorhill on purpose -- see - [gorhill/uMatrix#30](https://github.com/gorhill/uMatrix/issues/30)

To my understanding, what I suggest is not the same as gorhill mentioned in 2014.

My suggestion is offer both ways of contributing rules:  
1. Ruleset recipes  
2. Normal rules (which are exactly the same as what you see in "My rules", having the same syntax)

**Comment by uBlock-user** [comment]
Reopened for @gorhill to decide, what you're suggesting is a redesign of ruleset recipes and could be a grander one depending on what it requires to be done.

**Comment by ReporterX** [comment]
I did not express my suggestion well and caused confusion. Sorry about that. Let me try to explain again.

The current "Ruleset Recipe" is fine. Users can review before applying changes. It offers choices. You can provide more than 1 ruleset in the same website and let users pick one. However there are disadvantages. It requires a lot of user interaction which is bad to the masses. See "Privacy Badge" and how many people are using it.

My suggestion is NOT to replace "Ruleset Recipe". It has its own purposes. Keep it.

My suggestion is to offer a **new extra** way to contribute and share rules. They can complement each other.

There will be two ways: 
1. Ruleset recipes
2. Normal rules

To be clear, what "normal rules" mean here is [something like this](https://github.com/uMatrix-Rules/uMatrix-Rules-Domain/blob/master/rules.txt). In other words, it is the same as what you see in "my rules".

### Why a new way to contribute rules? 

Imagine a capable person wants to share their quality collection of rules to the rest of the world.  It contains about 5000 thousand lines. Under the current mode, they have to rewrite 5000 thousand lines to "Ruleset Recipe" format before they can share.

Under the new way, they just need to submit the current rules. Much simpler and easier, right?

The masses want "set and forget". "Ruleset Recipe" cannot fulfil such needs.

"Normal rules" is one click subscription. Once a user decides to subscribe that list, it is "set and forget". It helps users to unbreak sites which the list covers, while enjoying security and privacy.

I believe uMatrix will expand much faster if we offer a new way to contribute rules.

**Comment by uBlock-user** [comment]
> Imagine a capable person wants to share their quality collection of rules to the rest of the world. It contains about 5000 thousand lines. Under the current mode, they have to rewrite 5000 thousand lines to "Ruleset Recipe" format before they can share.

FYI, Ruleset recipes is not the only way for users to import rules. Download the [txt](https://github.com/uMatrix-Rules/uMatrix-Rules-Domain/blob/master/rules.txt) file and then open my Rules in uMatrix Dashboard, then click on import followed by commit and you're done.

**Comment by ReporterX** [comment]
I created a table. Both mode have different purpose and can fulfill different needs.

![Two mode compare](https://user-images.githubusercontent.com/8235338/56077937-ac06e280-5e14-11e9-8012-12d86d4010a4.png)

> 
> > Imagine a capable person wants to share their quality collection of rules to the rest of the world. It contains about 5000 thousand lines. Under the current mode, they have to rewrite 5000 thousand lines to "Ruleset Recipe" format before they can share.
> 
> FYI, Ruleset recipes is not the only way for users to import rules. Download the [txt](https://github.com/uMatrix-Rules/uMatrix-Rules-Domain/blob/master/rules.txt) file and then open my Rules in uMatrix Dashboard, then click on import followed by commit and you're done.

Thank you for your information. I know how to import rules. 

I hope my suggestion could benefit everyone, not just me.

We need a subscription list (just like Hosts files) so people can subscribe to different lists to unbreak many common sites in one click at ease.

**Comment by gorhill** [comment]
> We need a subscription list (just like Hosts files) so people can subscribe to different lists to unbreak many sites automatically at ease.

Duplicate of <https://github.com/gorhill/uMatrix/issues/400>.
